playground/src/pages/primitives.vue

```diff
@@ -0,0 +1,103 @@
+<script setup lang="ts">
+import { ref, watchEffect } from 'vue'
+import { 
+  BasicShadowMap,
+  SRGBColorSpace,
+  NoToneMapping,
+  Mesh,
+  TorusGeometry,
+  MeshToonMaterial,
+  TorusKnotGeometry,
+  PlaneGeometry,
+  Group,
+  SphereGeometry, 
+} from 'three'
+import { TresCanvas } from '@tresjs/core'
+import { OrbitControls } from '@tresjs/cientos'
+import { TresLeches, useControls } from '@tresjs/leches'
+import '@tresjs/leches/styles'
+
+const gl = {
+  clearColor: '#82DBC5',
+  shadows: true,
+  alpha: false,
+  shadowMapType: BasicShadowMap,
+  outputColorSpace: SRGBColorSpace,
+  toneMapping: NoToneMapping,
+}
+const canvas = ref()
+const meshRef = ref()
+
+const { knot } = useControls({
+  knot: false,
+})
+
+const { isVisible } = useControls({
+  isVisible: true,
+})
+
+watchEffect(() => {
+  if (meshRef.value) {
+    console.log(meshRef.value)
+  }
+})
+
+const torus = new Mesh(
+  new TorusGeometry(1, 0.5, 16, 100),
+  new MeshToonMaterial({
+    color: '#82DBC5',
+  }),
+)
+
+const torusKnot = new Mesh(
+  new TorusKnotGeometry(1, 0.5, 100, 16),
+  new MeshToonMaterial({
+    color: '#ff00ff',
+  }),
+)
+
+const sphere = new Mesh(
+  new SphereGeometry(1, 32, 32),
+  new MeshToonMaterial({
+    color: '#82DBC5',
+  }),
+)
+
```

Comment:
In `nodeOps.ts`, I mentioned that cloning means that setup references are lost. Below is some demonstration code.

Both `torus` and `torusKnot` continue to exist after the `clone`, but they no longer point to the on-screen objects, so the on-screen objects don't update.

If the objects aren't `clone()`d in `nodeOps`, the on-screen shapes will rotate – but other errors pop up.

```
useRenderLoop().onLoop(() => {
  torus.rotateX(0.01)
  torusKnot.rotateY(0.01)
})
```

I don't know if there's a workaround here. Just pointing it out.

Reply:
@andretchen0 This argument makes me think we should abandon the idea of trying to make `:object` prop reactive, swiping instances without losing the ref is getting too complicated.

Users could still use primitives with conditional rendering, I wanted `primitive` to work similar to `component` but honestly at this point I don't know how to achieve it.

Reply:
@alvarosabu

I don't understand the internals of Tres well enough to have an opinion on the feasibility here. 

I'll start reading the source after I finish `<AnimatedSprite />` for Cientos.

>  I wanted primitive to work similar to component but honestly at this point I don't know how to achieve it.

I agree that this is a worthwhile goal. It'd be great if it "just works".

Reply:
@alvarosabu

Sure thing. That could be fun!

Reply:
Thanks! I'll go read the Discord.